### PR TITLE
feat(Config-v2): Support for optlyconfig-v2 added.

### PR DESCRIPTION
**Pull request by yasirfolio3**
## Summary
- Support for OptimizelyConfig v2 added.

## Issues
- All Config-v2 tests should pass.

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -12,7 +12,7 @@ require (
 	github.com/go-kit/kit v0.9.0
 	github.com/google/uuid v1.1.1
 	github.com/lestrrat-go/jwx v0.9.0
-	github.com/optimizely/go-sdk v1.6.2-0.20210507191537-aba050a09690
+	github.com/optimizely/go-sdk v1.6.2-0.20210903184112-9d97c16f5787
```
Comment (msohailhussain) [inline go.mod:15]: this should be update with the go release.

**Review by msohailhussain — COMMENTED** [review]
looks fine to me. leaving it on matjaz for acceptance and one more review from jae/jacob

File: api/openapi-spec/openapi.yaml [hunk 2]
```diff
@@ -306,23 +306,57 @@ components:
...
     Feature:
       properties:
         id:
           type: string
         key:
           type: string
+        experimentRules:
+          type: array
+          items:
+            $ref: '#/components/schemas/Experiment'
```
Comment (The-inside-man) [inline api/openapi-spec/openapi.yaml:351]: Should these be OptimizelyExperiments or does it follow the same as Experiment from config? Same for feature and attribute.

File: tests/acceptance/test_acceptance/test_batch.py [hunk 3]
```diff
@@ -47,36 +47,78 @@ def test_batch_valid_reponse(session_obj):
...
+					"key": "15439080355",
+					"variationsMap": {
+						"15404730339": {
+							"featureEnabled": true,
+							"id": "15404730339",
+							"key": "15404730339",
+							"variablesMap": {
+								"fff": {
+									"id": "15427520260",
+									"key": "fff",
```
Comment (Mat001) [inline tests/acceptance/test_acceptance/test_batch.py:94]: Just curious, did you have to add these variables in the app.optmizely.com project or were they already there? (I created the project but I forgot)

File: tests/acceptance/test_acceptance/test_config.py [hunk 4]
```diff
@@ -7,121 +7,322 @@
...
-                },
-                "variation_2": {
-                    "id": "16927770169",
-                    "key": "variation_2",
-                    "featureEnabled": false,
-                    "variablesMap": {}
-                }
-            }
+  "environmentKey": "",
+  "sdkKey": "",
```
Comment (Mat001) [inline tests/acceptance/test_acceptance/test_config.py:11]: Hmm, here we're getting empty values. I know why, it's because it's already an existing project and the flag wasn't turned on. I wonder if we should modernize our project for Agent acceptance tests. Currently it uses legacy UI and doesn't include sdk and env keys.
maybe I can create a Jira ticket. And then we update this test so we will have sdk and environment key values in.
What do you think @msohailhussain

Comment (Mat001) [reply tests/acceptance/test_acceptance/test_config.py:11]: As a minimum this test works, the keys show, but with empty values. At least the keys can be verified.

**Review by Mat001 — APPROVED** [review]
LGTM. Can you please respond about my comment about updating the app.optmizely project for acceptance tests. thx

File: api/openapi-spec/openapi.yaml [hunk 5]
```diff
@@ -470,7 +526,7 @@ components:
         type:
           type: string
         value:
-          type: object
+          type: string
```
Comment (jaeopt) [inline api/openapi-spec/openapi.yaml:529]: why this change? JSON type?

**Review by jaeopt — CHANGES_REQUESTED** [review]
Clarification and suggestions for some naming changes

File: tests/acceptance/test_acceptance/test_batch.py (hunk 3 above)
Comment (yasirfolio3) [reply tests/acceptance/test_acceptance/test_batch.py:94]: they were already there.

File: api/openapi-spec/openapi.yaml (hunk 5 above)
Comment (yasirfolio3) [reply api/openapi-spec/openapi.yaml:529]: It is of type `string` in go-sdk. Guess it was wrong earlier. https://github.com/optimizely/go-sdk/blob/master/pkg/config/optimizely_config.go#L106

File: go.mod (hunk 1 above)
Comment (yasirfolio3) [reply go.mod:15]: reverting it will cause the unit tests to fail.

File: api/openapi-spec/openapi.yaml [hunk 6]
```diff
@@ -315,7 +315,7 @@ components:
       required:
         - id
         - key
-    Attribute:
+    OptimizelyAttribute:
```
Comment (jaeopt) [inline api/openapi-spec/openapi.yaml:318]: It looks ok but want to confirm that theres is no implications of these renaming to existing agent users.

**Review by jaeopt — CHANGES_REQUESTED** [review]
Looks good. One more name change suggested.

File: api/openapi-spec/openapi.yaml (hunk 6 above)
Comment (yasirfolio3) [reply api/openapi-spec/openapi.yaml:318]: Since we have only updated the model names and haven't removed/renamed any old `key/value` pair in the optimizelyConfig, there shouldn't be any implication i believe. What do you think? @msohailhussain

**Review by jaeopt — APPROVED** [review]
LGTM

**Review by jaeopt — APPROVED** [review]
LGTM
